vfs-azure/src/main/java/com/sludev/commons/vfs2/provider/azure/AzFileProvider.java

```diff
@@ -123,100 +122,91 @@
         Capability.DELETE
     }));
 
-    private String endpoint;
-    
-    /**
-     * Set the S3 endpoint we should use.  This needs to be done before init() is called if done at all.
-     * 
-     * The value is optional, hence best to leave undeclared.
-     * 
-     * @param ep The optional endpoint E.g. <account>..blob.core.windows.net 
-     */
-    public void setEndpoint(String ep)
-    {
-        endpoint = ep;
-    }
-    
+
     /**
      * Construct a new provider for use with a File-System Manager object.
      */
-    public AzFileProvider()
-    {
+    public AzFileProvider() {
         super();
         setFileNameParser(AzFileNameParser.getInstance());
     }
-    
+
+
+
     /**
      * In the case that we are not sent FileSystemOptions object, we need to have
      * one handy.
-     * 
-     * @return 
+     *
+     * @return
      */
-    public static FileSystemOptions getDefaultFileSystemOptions()
-    {
+    public static FileSystemOptions getDefaultFileSystemOptions() {
         return DEFAULT_OPTIONS;
     }
-    
+
+
     /**
-     * Callback for handling the create File-System event
-     * 
+     * Callback for handling the create file-System event
+     *
      * @param rootName
      * @param fileSystemOptions
      * @return
-     * @throws FileSystemException 
+     * @throws FileSystemException
      */
     @Override
-    protected FileSystem doCreateFileSystem(FileName rootName, FileSystemOptions fileSystemOptions) throws FileSystemException
-    {
+    protected FileSystem doCreateFileSystem(FileName rootName, FileSystemOptions fileSystemOptions)
+            throws FileSystemException {
+
+        AzFileName fileName = (AzFileName) rootName;
```

Comment:
optional: we could call this `rootFileName` or `azRootName`.  But happy for this to be left as-is if you prefer.